src/bio/bio_wal.c

```diff
@@ -1645,6 +1659,24 @@ bio_wal_replay(struct bio_meta_context *mc,
 		D_DEBUG(DB_IO, "Replayed %u WAL transactions\n", nr_replayed);
 		D_ASSERT(wal_id_cmp(si, tx_id, si->si_commit_id) > 0);
 		si->si_unused_id = wal_next_id(si, si->si_commit_id, si->si_commit_blks);
+
+		/* create and set replay metrics */
+		clock_gettime(CLOCK_REALTIME, &tms);
+		e_us = (tms.tv_sec * 1000000) + (tms.tv_nsec / 1000);
+		uuid_unparse_lower(mc->mc_wal->bic_pool_id, pool);
+		rc = d_tm_add_metric(&wal_sz, D_TM_GAUGE, "WAL replay size", "bytes",
+				     "pool/%s/tgt_%d/wal_replay/size", pool, trg);
```

Comment:
hmm, how can we create metric on each replay? 

It's bit difficult to make per pool replay metrics, but I think we don't have to make it per pool, we could make it per engine like other metrics in BIO, so the metrics could be created in dma_metrics_init().

Reply:
I agree that per pool metrics is better than per engine, to make it per pool, you could refer to dss_module_metrics and how other modules generating per pool metrics.

Reply:
As far as I understand, the replay happens only once per life of a pool (if it happens at all) so  I'm not sure what is the point of creating a statistical metric. It will only have one value. Correct me if I'm wrong here.
As for capturing statistical distribution of replay time across multiple pools/targets, we might have a not-so-trivial problem here. These replay are happening in parallel while yielding the execution thread to each other. Attempts to calculate average/mean values may be tricky. We can still add this "statistical" metric if it is required, but I think we should leave the one discussed here as is as it give the actual time spent in replay per pool/target.